src/App.tsx

```diff
@@ -1,109 +1,26 @@
 import React from 'react';
-
 import './App.scss';
+import postsFromServer from './api/posts';
+import commentsFromServer from './api/comments';
+import usersFromServer from './api/users';
+import { Post } from './types/Post';
+import { PostList } from './components/PostList';
+
+type InitialPosts = Omit<Post, 'user' | 'comments'>;
+
+function getInformativePosts(initialPosts: InitialPosts[]): Post[] {
+  return initialPosts.map(post => ({
+    ...post,
+    user: usersFromServer.find(userFind => userFind.id === post.userId) || null,
```

Comment:
Let's decompose your logic
Move find user & filter comments to the separate fucntions